client/src/main.tsx

```diff
@@ -1,17 +1,38 @@
 import React from "react";
 import ReactDOM from "react-dom/client";
 
-import { Login, Register, Test } from "./components/pages";
+import { Login, Register, Test, Loading } from "./components/pages";
 
-import { createBrowserRouter, RouterProvider } from "react-router-dom";
+import { createBrowserRouter, RouterProvider, redirect, Outlet } from "react-router-dom";
 
-import PrivateRoute from "./auth/PrivateRoute.tsx";
 import { ChakraProvider } from "@chakra-ui/react";
+import { userProfileSchema } from "./schemas.ts";
 
 const router = createBrowserRouter([
   {
     path: "/",
-    element: <PrivateRoute />,
+    loader: async () => {
+      try {
+        const response = await fetch("localhost:8080/implicit_login", {
+          mode: "cors",
```

Comment:
nitpick: I'm not sure why we need this field. Is there anything I'm not noticing on why we have set `mode` to "cors" here?